src/core/geometry_utils.h

```diff
@@ -8,50 +14,117 @@
 #define EQ_DOUBLE(a, b)                         \
   (std::abs((a) - (b)) < 0.0001)
 
+/**
+ * \brief  Structure describes a simple horizontal rectangle
+ */
+
 struct Rectangle {
+
+  /**
+   * Default constructor
+   * Creates a rectangle with zero area in point (0,0)
+   */
   Rectangle() : Rectangle(0, 0, 0, 0) {}
+
+  /**
+   * Constructor with parameters.
+   * Creates a rectangle where all contained points are bounded in limits l < x < r, b < y <t
+   * \param b Bottom of a rectangle
+   * \param t Top of rectangle
+   * \param l Left side of rectangle
+   * \param r Right side of rectangle
+   */
   Rectangle(double b, double t, double l, double r) :
     bot(b), top(t), left(l), right(r) {}
 
+  /**
+   * Calculates whether a rectangle contains a point
+   * \param x,y Coordinates of a target point
+   * \return Returns True if rectangle contains a point and False otherwise
+   */
   bool does_contain(double x, double y) const {
     return ((bot < y) && (y < top)) && ((left < x) && (x < right));
   }
 
+  /**
+   * Calculates an area of a rectangle
+   * \return Value of an area
+   */
   double area() const {
     return (top - bot)*(right - left);
   }
 
-  double bot, top, left, right;
+  double bot,	///< Bottom of a rectangle
+  	  	 top,	///< Top of a rectangle
+		 left,	///< Left side of a rectangle
+		 right;	///< Right size of a rectangle
 };
 
+/**
+ * \brief Structure describes a simple point in a plane
```

Comment:
"describes a simple" -> "defines a";
"point in a plane" -> "point on a plane" (?)